torch/nn/modules/container.py

```diff
@@ -97,11 +106,17 @@ def __init__(self, modules=None):
             self += modules
 
     def __getitem__(self, idx):
-        if not (-len(self) <= idx < len(self)):
-            raise IndexError('index {} is out of range'.format(idx))
-        if idx < 0:
-            idx += len(self)
-        return self._modules[str(idx)]
+        if isinstance(idx, int):
+            if not (-len(self) <= idx < len(self)):
+                raise IndexError('index {} is out of range'.format(idx))
+            if idx < 0:
+                idx += len(self)
+            items = self._modules.items()
+            if not isinstance(items, list):
+                items = list(items)
+            return items[idx][1]
+        else:
+            return self._modules[idx]
```

Comment:
These modules don't have usable names. This class is meant to behave exactly like Python lists, and they can't be indexed with arbitrary objects